File: algorithms/1_easy/N/02367_number_of_arithmetic_triplets/bruteforce.rs
impl Solution {
    pub fn arithmetic_triplets(nums: Vec<i32>, diff: i32) -> i32 {
        let len = nums.len();
        let mut count = 0_i32;

        for i in 0..len {
            for j in (i + 1)..len {
                for k in (j + 1)..len {
                    if nums[k] - nums[j] == diff && nums[j] - nums[i] == diff {
                        count += 1;
                    }
                }
            }
        }

        count
    }
}

File: algorithms/1_easy/C/01995_count_special_quadruplets/bruteforce.py
class Solution:
    def countQuadruplets(self, nums: List[int]) -> int:
        n = len(nums)
        count = 0

        for d in range(n - 1, -1, -1):
            for c in range(d - 1, -1, -1):
                for b in range(c - 1, -1, -1):
                    for a in range(b - 1, -1, -1):
                        if nums[d] == nums[a] + nums[b] + nums[c]:
                            count += 1

        return count
